### PR TITLE
Improving results plot

### PR DESCRIPTION
**Pull request by jeandut**
This PR is aimed at prettifying the plot of results of the benchmark in two ways:
- Fixing the display of the current plot for when the y axis is shared on each row
- Adding an option that removes the sharing of the y-axis and add metric labels (SHAREY=True/False)

### CODE REVIEW

File: flamby/results/plot_results.py [hunk 1]
```diff
@@ -164,9 +189,12 @@ def get_nb_clients_from_dataset(name_arg):
     current_title = title_dicts[name]
     ax.set_title(current_title, fontsize=45, fontweight="heavy")
 
-# Hide the last plot
+# Hide the two plots on the last row
 flattened_axs[-3].set_visible(False)
 flattened_axs[-1].set_visible(False)
+
+# No idea why I need to add that probably because of sharey on the last row
```
Comment (maikia) [inline flamby/results/plot_results.py:196]: this comment is about previous or following lines?

**Review by maikia — APPROVED** [review]
LGTM. Thanks for your contribution ;-) 
For the next time:
1. Can you write at least one liner description of what your PR is doing (Improving results plot in the title isn't very explicit)
2. Can you use conventional commit?

File: flamby/results/plot_results.py (hunk 1 above)
Comment (jeandut) [reply flamby/results/plot_results.py:196]: Good catch !

Comment (jeandut) [inline flamby/results/plot_results.py:196]:
```suggestion
```

File: flamby/results/plot_results.py [hunk 2]
```diff
@@ -164,9 +189,11 @@ def get_nb_clients_from_dataset(name_arg):
     current_title = title_dicts[name]
     ax.set_title(current_title, fontsize=45, fontweight="heavy")
 
-# Hide the last plot
+# Hide the two plots on the last row
 flattened_axs[-3].set_visible(False)
 flattened_axs[-1].set_visible(False)
+
```
Comment (jeandut) [inline flamby/results/plot_results.py:195]:
```suggestion
```

File: flamby/results/plot_results.py [hunk 3]
```diff
@@ -164,9 +189,10 @@ def get_nb_clients_from_dataset(name_arg):
...
```
Comment (jeandut) [inline flamby/results/plot_results.py:195]:
```suggestion
```

**Pull request merged**